seismicpro/survey/metrics.py

```diff
@@ -29,3 +66,780 @@ def plot(self, ax, coords, index, sort_by=None, **kwargs):
     def get_views(self, sort_by=None, **kwargs):
         """Return a single view, that plots a gather sorted by `sort_by` by click coordinates."""
         return [partial(self.plot, sort_by=sort_by)], kwargs
+
+
+class TracewiseMetric(SurveyAttribute):
+    """Base class for tracewise metrics with additional plotters and aggregations. Child classes should redefine
+    `get_mask` or `numba_get_mask` and `description` methods, and optionally `preprocess`."""
+    threshold = None
+    top_ax_y_scale = "linear"
+
+    def __call__(self, gather):
+        """Compute qc metric by applying `self.preprocess`, `self.get_mask` and `self.aggregate` to provided gather."""
```

Comment:
You describe code in words
```suggestion
        """Compute qc metric. """
```

Reply:
discussed in person, left as this